src/core/dbus.c

```diff
@@ -977,7 +977,7 @@ static int bus_init_private(Manager *m) {
                 char *p = sa.un.sun_path;
                 const char *e;
 
-                e = secure_getenv("XDG_RUNTIME_DIR");
+                e = getenv("XDG_RUNTIME_DIR");
```

Comment:
The fact that sd-bus is used and use secure_getenv() is used everywhere will indeed defeat my proposed patch, I found indeed one other occurrence of `secure_getenv()` in the code path while testing it further.